### PR TITLE
Fix dummy process unable to terminate in test_process_checker

### PR DESCRIPTION
**Pull request by ichorid**
The dummy process does not react to `terminate` and only dies with the interpreter. This patch adds a correct, `Value`-based shared var mechanism to shut it down.

### CODE REVIEW

**Comment by devos50** [comment]
@ichorid please rebase on the latest commit on the next branch; it seems you have some of my work in this PR.

**Comment by ichorid** [comment]
yup, already noticed that...

File: Tribler/Test/Core/Modules/test_process_checker.py [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 
 from Tribler.Core.Modules.process_checker import ProcessChecker, LOCK_FILE_NAME
 from Tribler.Test.test_as_server import AbstractServer
 
 
-def process_dummy_function():
-    while True:
+def process_dummy_function(stop_flag):
+    while stop_flag.value == 0:
+        sleep(0.001)
```
Comment (qstokkink) [inline Tribler/Test/Core/Modules/test_process_checker.py]: Granularity of sleep is only in 10's of ms (0.01).

Comment (ichorid) [reply Tribler/Test/Core/Modules/test_process_checker.py]: ow!

**Comment by qstokkink** [comment]
Please wait until the unit tests are finished before you merge.

**Pull request merged**